### PR TITLE
consensus: additional timing metrics 

### PR DESCRIPTION
**Pull request by williambanfield**
backport of: #9168

This metric is being added in relation to investigating database compaction performance. When testing database compaction, we'd like to understand if the compaction will significantly increase a consensus step time.

---

#### PR checklist

- [ ] Tests written/updated, or no tests needed
- [ ] `CHANGELOG_PENDING.md` updated, or no changelog entry needed
- [ ] Updated relevant documentation (`docs/`) and code comments, or no
      documentation updates needed

### CODE REVIEW

File: consensus/state.go [hunk 1]
```diff
@@ -520,6 +520,14 @@ func (cs *State) updateHeight(height int64) {
 }
 
 func (cs *State) updateRoundStep(round int32, step cstypes.RoundStepType) {
+	if !cs.replayMode {
+		if round != cs.Round || round == 0 && step == cstypes.RoundStepNewRound {
+			cs.metrics.MarkRound(cs.Round, cs.StartTime)
```
Comment (cmwaters) [inline consensus/state.go:525]: Is `cs.StartTime` the right time value to be comparing with? Isn't this only set when we update state to the next height and not after every round?

Also in `updateToState`, `cs.StartTime` is set _after_ `updateRoundStep` is called

File: consensus/metrics.go [hunk 2]
```diff
@@ -25,6 +29,9 @@ type Metrics struct {
 	// Number of rounds.
 	Rounds metrics.Gauge
 
+	// Histogram of round duration.
+	RoundDuration metrics.Histogram
```
Comment (cmwaters) [inline consensus/metrics.go:33]: I'm not sure how useful round duration is in regards to measuring the impact of database compaction

File: consensus/state.go (hunk 1 above)
Comment (williambanfield) [reply consensus/state.go:525]: Removed `RoundDuration` metric.

File: consensus/metrics.go (hunk 2 above)
Comment (williambanfield) [reply consensus/metrics.go:33]: Removed.

**Pull request merged**